### PR TITLE
Info section

### PR DESCRIPTION
**Pull request by dcwds**
The following changes have been made according to the AC:
- add info sections: how it works, about
- add id to sections so that they can be viewed via header links

Other changes:
- tweaks to a lot of link styles across the app
- overall font-size of app changes depending on device resolution

Some thoughts:
For header and footer components, I do select link children (`a`) to apply styling. Is this a reasonable assumption to make? I understand the value of decoupling styles from the DOM structure but is it reasonable in these cases?

When creating the `info-section` branch locally, I checked out from my local `hero-section` branch which contained changes that were not yet merged to `main`. I eventually merged my `hero-section` PR to `main`, checked out `main` locally and did `git pull origin main`, then checked out `info-section` and merged `main` into it. However, as you can see in this PR's commit history, those `hero-section` commits are still shown, despite the merge. Is this a case for `git rebase` so this doesn't occur?

### CODE REVIEW

**Comment by mentor-tara** [comment]
Great job :rocket: I'm a bit busy at the moment but I'll leave a review soon.

In the meantime, consider starting with the next task already to practice the professional Git workflow. [If you need a refresher, feel free to follow this section in the course again.](https://profy.dev/project/reddit-timer/while-waiting)[](ckmadiu6g3nlc0a84dmaspn6b)

**Review by mentor-tara — CHANGES_REQUESTED** [review]
Before I approve this PR I'd like to share a few tips with you.

**Please go through the list below and review your own code first.** Imagine you're the reviewer or a person who didn't see the code before. Ask yourself if everything is clearly written and understandable. Are components and variables properly named?

Reviewing your own code with this mindset is an important practice that will help you learn a lot and write high-quality code.

To make it a bit easier for you here are the tips I promised. Some have a link :link: attached where you can find more information. I'd recommend reading the **bold tips** in any case. Feel free to check the boxes when you think your code fits.

**General:**
- [ ] Use the `article` tag instead of `section` [:link:](https://profy.dev/project/reddit-timer/review-tips/prefer-article-over-section)

**Styles:**
- [ ] **Refrain from styling children with CSS selectors** [:link:](https://profy.dev/project/reddit-timer/review-tips/css-selectors)
- [ ] **Extract shared CSS values** [:link:](https://profy.dev/project/reddit-timer/review-tips/common-css-values-in-theme)
- [ ] Use global styles when appropriate (e.g. for font-size of headings, color of links...) [:link:](https://profy.dev/project/reddit-timer/review-tips/globally-style-elements)

**Tests:**
- [ ] **Use [within](https://testing-library.com/docs/dom-testing-library/api-helpers#within-and-getqueriesforelement-apis) to test multiple occurences of elements (e.g. the profy.dev link)** [:link:](https://profy.dev/project/reddit-timer/review-tips/testing-library-within)

Once you're done with reviewing your own code feel free to check the [example implementation](https://profy.dev/project/reddit-timer/code-examples/info-section-54645480-9f74-4a7a-b1c0-a98de4445a2a). **But remember: It's an important exercise to review your code first on your own.**

When you think this PR is ready to be merged just let me know by requesting another review. I'll approve right away.[](ckmbnzw4843z70b41h7aujv0o)

**Review by mentor-tara — APPROVED** [review]
Great work! :tada::tada::tada: You can merge the PR now.[](ckmadn6tc3o1v0b87a2kyd9yb)

### Note: This is the last task included in the free membership.

It's up to you of course, but I'd be super happy to finish this project with you.

**_This is what you'll get when you upgrade to a premium membership:_**

- Get access to the tasks and designs for the search page.
- Build a dynamic form to fetch data from an API.
- Transform the data as preparation for the UI.
- Build the interactive heatmap.
- Write automated tests for dynamic components.
- Get advice on how to prepare this project for your job hunt.

I hope you're ready to take the leap. This is where it gets real.

# [Go Premium Now](https://profy.dev/#premium)

See you soon 😄 

> Note: If you're already a premium member, you can continue right away, of course.[](ckmadn6tc3o1v0b87a2kyd9yb)

**Comment by jkettmann** [comment]
You're right, this is a case for rebasing this branch on the main branch with the interactive option. While you're on the `info-section` branch you can run the following command. An editor will open in the terminal with a list of commits. Simply remove all the commits **that don't belong to this branch** from the list.

```
git rebase main -i
```

Please make sure to push the branch first if you want to try this. This command changes the Git history and you might lose you changes if something goes wrong. By pushing the branch first you have a backup from which you can recover.

File: src/components/header/header.test.js [hunk 1]
```diff
@@ -4,39 +4,26 @@ import Header from './header';
...
 
-    expect(spyHistoryPush).toHaveBeenCalledWith('/#how-it-works');
+    expect(getByRole('link', { name: /how it works/i })).toHaveAttribute('href', '/#how-it-works');
   });
 
   it('routes about link', () => {
-    const { getByRole } = header;
-
-    fireEvent.click(getByRole('link', { name: /about/i }));
+    const { getByRole } = render(<Header />);
```
Comment (jkettmann) [inline src/components/header/header.test.js:14]: You can import `screen` from `@testing-library/react` and use `screen.getByRole` instead. This is valid for all the queries. The advantage is that you have auto-completion in your editor and you don't need to update this line every time you need to use another query.

File: src/constants.js [hunk 2]
```diff
@@ -1 +1,3 @@
 export const DEFAULT_SUBREDDIT = 'javascript';
+export const PROFY_LINK = 'https://profy.dev';
+export const PROFY_EMPLOYERS_LINK = 'https://profy.dev/employers';
```
Comment (jkettmann) [inline src/constants.js:3]: Nice abstraction

File: src/components/info/info.style.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    margin-right: 0.313rem; /* 5px */
+  }
+`;
+
+export const Link = styled.a`
+  color: ${(props) => props.theme.color.link.default[0]};
+
+  &:hover {
+    color: ${(props) => props.theme.color.link.default[1]};
+  }
```
Comment (jkettmann) [inline src/components/info/info.style.js:47]: These are the default styles for anchors, right? In that case you can move these styles to the global styles and target the `a` tag

File: src/styles/theme.js [hunk 4]
```diff
@@ -25,11 +25,12 @@ const theme = {
     default: '#93918F',
     headline: '#000000',
     link: {
+      default: ['#0087FF', '#1360A4'],
```
Comment (jkettmann) [inline src/styles/theme.js:28]: Using an array here makes it hard to understand tbh. Only from reading the component styles it becomes clear that the second is the hover color.

File: src/components/info/info.style.js (hunk 3 above)
Comment (dcwds) [reply src/components/info/info.style.js:47]: The problem with declaring them globally is bad inheritance starts to occur on things like the CTA button which uses an `a` tag (and doesn't want the color to change on hover) and I begin fighting off those styles by having to explicitly declare color states for other components that happen to include `a` tags, if we're assuming a scenario where this project is expanded.

Comment (jkettmann) [reply src/components/info/info.style.js]: That sounds normal to me tbh. A link that looks like a button will have to overwrite some of the default link styles. I'm not sure if there would be so many other scenarios like this. Maybe something like card elements that are wrapped with an anchor and contain text. But also there I'd say it's expected that default styles need to be overwritten (think of the typical `text-decoration: underline` that's often set by default for links)

**Pull request merged**